client/src/main/kotlin/io/spine/examples/shareaware/client/Client.kt

```diff
@@ -0,0 +1,237 @@
+            }
+    }
+
+    private val client: Client
+    private lateinit var user: UserId
+    private lateinit var walletId: WalletId
+
+    init {
+        val channel = ManagedChannelBuilder
+            .forAddress(host, port)
+            .usePlaintext()
+            .build()
+        client = Client
+            .usingChannel(channel)
+            .build()
+    }
+
+    /**
+     * Sends command to authenticate user in the system.
+     *
+     * A command `CreateWallet` is used for user creation because in the ShareAware
+     * there is no functional to register or authenticate the user, yet.
+     */
+    public fun authenticateUser() {
+        val userId = UUID.randomUUID().toUserId()
+        val walletId = userId.toWalletId()
+        val walletField = WalletCreated.Field.wallet()
+        val walletCreated: CompletableFuture<WalletCreated> = CompletableFuture()
+        this.subscribeToEvent(
+            WalletCreated::class.java,
+            EventFilter.eq(walletField, walletId)
+        )
+        {
+            walletCreated.complete(it)
+        }
+        command(createWallet(walletId))
+        if (!::user.isInitialized) {
+            val event = walletCreated.get()
+            user = event.wallet.owner
+            this.walletId = event.wallet
+        }
+    }
+
+    /**
+     * Sends a command to the server.
+     */
+    public fun command(message: CommandMessage) {
+        clientRequest()
+            .command(message)
+            .postAndForget()
+    }
+
+    /**
+     * Subscribes to the changes in entity of the given type.
+     *
+     * @param type type of the entity on which changes subscription works
+     * @param id entity ID by which arrived entities will be filtered
+     * @param observer callback function that will be triggered when the entity state changes
+     */
+    public fun <S : EntityState> subscribeToEntity(type: Class<S>, id: Message, observer: (S) -> Unit) {
+        clientRequest()
+            .subscribeTo(type)
+            .byId(id)
+            .observe(observer)
+            .post()
+    }
+
+    /**
+     * Subscribes to the events of the provided type.
+     *
+     * @param type type of the event to subscribe on
+     * @param filter filter by which arrived events of the provided type will be filtered
+     * @param observer callback function that will be triggered when the event arrives
+     */
+    public fun <E : EventMessage> subscribeToEvent(
+        type: Class<E>,
+        filter: EventFilter,
+        observer: (E) -> Unit
+    ) {
+        clientRequest()
+            .subscribeToEvent(type)
+            .where(filter)
+            .observe(observer)
+            .post()
+    }
+
+    /**
+     * Retrieves the entity with a provided type and ID.
+     *
+     * @param type type of the entity to be retrieved
+     * @param id entity ID by which the query result will be filtered
+     */
+    public fun <E : EntityState> readEntity(type: Class<E>, id: Message): E? {
```

Comment:
Let's document the returning value as well.

That's a part of a standard set of rules: public API should be documented. This is a part of public API. Please review this changeset for that.